app/src/test/java/com/techcourse/DispatcherServletTest.java

```diff
@@ -0,0 +1,98 @@
+package com.techcourse;
+
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.io.IOException;
+
+import jakarta.servlet.RequestDispatcher;
+import jakarta.servlet.ServletException;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import jakarta.servlet.http.HttpSession;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Test;
+
+public class DispatcherServletTest {
+
+    private DispatcherServlet sut;
+    private HttpServletRequest request;
+    private HttpServletResponse response;
+
+    @BeforeEach
+    void setUp() {
+        sut = new DispatcherServlet();
+        sut.init();
+        request = mock(HttpServletRequest.class);
+        response = mock(HttpServletResponse.class);
+    }
+
+    @Nested
+    @DisplayName("Controller 인터페이스 기반 컨트롤러에 대한 요청을 처리한다.")
+    class ControllerInterface {
+        @Test
+        @DisplayName("login에 대한 요청을 처리한다.")
+        void handleLogin() throws IOException, ServletException {
+            // given
+            final var session = mock(HttpSession.class);
+            final var requestDispatcher = mock(RequestDispatcher.class);
+
+            when(request.getParameter("account")).thenReturn("gugu");
+            when(request.getParameter("password")).thenReturn("password");
+            when(request.getRequestURI()).thenReturn("/login");
+            when(request.getMethod()).thenReturn("POST");
+            when(request.getSession()).thenReturn(session);
+            when(request.getRequestDispatcher(anyString())).thenReturn(requestDispatcher);
+
+            // when
+            sut.service(request, response);
+
+            // then
+            verify(response).sendRedirect("/index.jsp");
+        }
```

Comment:
```suggestion
        @Test
        @DisplayName("login에 대한 요청을 처리한다.")
        void handleLogin() throws IOException, ServletException {
            // given
            final var session = mock(HttpSession.class);

            when(request.getParameter("account")).thenReturn("gugu");
            when(request.getParameter("password")).thenReturn("password");
            when(request.getRequestURI()).thenReturn("/login");
            when(request.getMethod()).thenReturn("POST");
            when(request.getSession()).thenReturn(session);

            // when
            sut.service(request, response);

            // then
            verify(response).sendRedirect("/index.jsp");
        }
```
여기선 포워딩을 하는지 검증하지 않으므로 requestDispatcher 관련 부분은 지워도 될 것 같아요 !